meeshkan/serve/mock/server.py

```diff
@@ -58,20 +30,50 @@ def __init__(
         routing=PathRouting(),
         log_dir: Optional[str] = None,
     ):
-        self._callback_dir = callback_dir
         self._admin_port = admin_port
         self._port = port
         self._specs = specs
         self._routing = routing
         self._scope = scope or Scope()
         self._log = Log(self._scope, NoSink() if log_dir is None else FileSink(log_dir))
 
+        self._mock_data_store = MockDataStore()
+        self._rest_middleware_manager = RestMiddlewareManager(self._mock_data_store)
+        self._callback_manager = callback_manager
+
+        if callback_dir is not None:
+            callback_manager.load(callback_dir)
+
+        self._request_processor = RequestProcessor(
```

Comment:
For me it would maybe be clearer if the specs were added to the mock data store here, maybe.

Reply:
Though I see one could then end up in inconsistent states, not sure what the correct solution would be.

Reply:
What kind of inconsistent state? I moved mock_data initialization upwards.